internal/cloud/awscloud/awscloud.go

```diff
@@ -385,61 +388,47 @@ func (a *AWS) CopyImage(name, ami, sourceRegion string) (string, error) {
 		return "", err
 	}
 
-	dIInput := &ec2.DescribeImagesInput{
-		ImageIds: []*string{result.ImageId},
-	}
-
-	// Custom waiter which waits indefinitely until a final state
-	w := request.Waiter{
-		Name:        "WaitUntilImageAvailable",
-		MaxAttempts: 0,
-		Delay:       request.ConstantWaiterDelay(15 * time.Second),
-		Acceptors: []request.WaiterAcceptor{
-			{
-				State:   request.SuccessWaiterState,
-				Matcher: request.PathAllWaiterMatch, Argument: "Images[].State",
-				Expected: "available",
-			},
-			{
-				State:   request.FailureWaiterState,
-				Matcher: request.PathAnyWaiterMatch, Argument: "Images[].State",
-				Expected: "failed",
-			},
-		},
-		Logger: a.ec2.Config.Logger,
-		NewRequest: func(opts []request.Option) (*request.Request, error) {
-			var inCpy *ec2.DescribeImagesInput
-			if dIInput != nil {
-				tmp := *dIInput
-				inCpy = &tmp
-			}
-			req, _ := a.ec2.DescribeImagesRequest(inCpy)
-			req.SetContext(aws.BackgroundContext())
-			req.ApplyOptions(opts...)
-			return req, nil
+	imgWaiter := ec2.NewImageAvailableWaiter(a.ec2)
+	imgWaitOutput, err := imgWaiter.WaitForOutput(
+		context.Background(),
+		&ec2.DescribeImagesInput{
+			ImageIds: []string{*result.ImageId},
 		},
-	}
-	err = w.WaitWithContext(aws.BackgroundContext())
+		time.Hour*24,
+	)
 	if err != nil {
 		return *result.ImageId, err
 	}
 
+	if imgWaitOutput.Images[0].State != ec2types.ImageStateAvailable {
+		return *result.ImageId, fmt.Errorf("Image not available after waiting: %s, Code: %v reason: %v",
+			imgWaitOutput.Images[0].State, *imgWaitOutput.Images[0].StateReason.Code, *imgWaitOutput.Images[0].StateReason.Message)
+	}
+
 	// Tag image with name
-	_, err = a.ec2.CreateTags(&ec2.CreateTagsInput{
-		Resources: []*string{result.ImageId},
-		Tags: []*ec2.Tag{
-			{
-				Key:   aws.String("Name"),
-				Value: aws.String(name),
+	_, err = a.ec2.CreateTags(
+		context.Background(),
+		&ec2.CreateTagsInput{
+			Resources: []string{*result.ImageId},
+			Tags: []ec2types.Tag{
+				{
+					Key:   aws.String("Name"),
+					Value: aws.String(name),
+				},
 			},
 		},
-	})
+	)
 
```

Comment:
(super nitpick) I would remove the `\n` here to have the error handling directly next to the error